### PR TITLE
adding ubuntu 24.04 LTS

### PR DESCRIPTION
**Pull request by samiura**
**Description:** added ubuntu-24.04

### CODE REVIEW

**Comment by jeffreyc-splunk** [comment]
Please add a `Dockerfile.ubuntu-noble` dockerfile in https://github.com/signalfx/splunk-otel-collector/blob/main/internal/buildscripts/packaging/tests/images/deb/ for testing.

**Comment by samiura** [comment]
> Please add a `Dockerfile.ubuntu-noble` dockerfile in https://github.com/signalfx/splunk-otel-collector/blob/main/internal/buildscripts/packaging/tests/images/deb/ for testing.
I do not see one for the latest ubuntu version. 
https://apt.puppetlabs.com/


Does this dockerfile need to change?

```
FROM ubuntu:24.04

RUN apt-get update &&\
    apt-get install -yq ca-certificates procps systemd wget apt-transport-https libcap2-bin curl gnupg lsb-release

ARG PUPPET_RELEASE="6"
RUN wget https://apt.puppetlabs.com/puppet${PUPPET_RELEASE}-release-noble.deb && \
    dpkg -i puppet${PUPPET_RELEASE}-release-noble.deb && \
    apt-get update && \
    apt-get install -y puppet-agent

RUN wget -O /tmp/nodejs.tar.gz https://nodejs.org/dist/v16.20.2/node-v16.20.2-linux-x64.tar.gz && \
    mkdir -p /opt/ && \
    tar -C /opt/ -xzf /tmp/nodejs.tar.gz && \
    mv /opt/node* /opt/node

ENV PATH=/opt/puppetlabs/bin:/opt/node/bin:$PATH

ENV container docker

RUN rm -f /lib/systemd/system/multi-user.target.wants/* \
    /etc/systemd/system/*.wants/* \
    /lib/systemd/system/local-fs.target.wants/* \
    /lib/systemd/system/sockets.target.wants/*udev* \
    /lib/systemd/system/sockets.target.wants/*initctl* \
    /lib/systemd/system/systemd-update-utmp*

RUN systemctl set-default multi-user.target
ENV init /lib/systemd/systemd

RUN puppet module install puppetlabs-stdlib --version 4.24.0
RUN puppet module install puppetlabs-apt --version 7.0.0

COPY deployments/puppet /etc/puppetlabs/code/environments/production/modules/splunk_otel_collector

VOLUME [ "/sys/fs/cgroup" ]

ENTRYPOINT ["/lib/systemd/systemd"]
```

**Comment by jeffreyc-splunk** [comment]
> @jeffreyc-splunk
> 
> > Please add a `Dockerfile.ubuntu-noble` dockerfile in https://github.com/signalfx/splunk-otel-collector/blob/main/internal/buildscripts/packaging/tests/images/deb/ for testing.
> > I do not see one for the latest ubuntu version.

Try just copying https://github.com/signalfx/splunk-otel-collector/blob/main/internal/buildscripts/packaging/tests/images/deb/Dockerfile.ubuntu-focal and changing `ubuntu:focal` to `ubuntu:noble`.

File: internal/buildscripts/packaging/tests/installer_test.py [hunk 1]
```diff
@@ -206,7 +206,7 @@ def fluentd_supported(distro, arch):
         return False
     elif distro == "amazonlinux-2023":
         return False
-    elif distro in ("debian-stretch", "ubuntu-xenial") and arch == "arm64":
+    elif distro in ("debian-stretch", "ubuntu-xenial", "ubuntu-noble") and arch == "arm64":
```
Comment (jeffreyc-splunk) [inline internal/buildscripts/packaging/tests/installer_test.py:209]:
```suggestion
    elif distro in ("debian-stretch", "ubuntu-xenial") and arch == "arm64":
        return False
    elif distro == "ubuntu-noble":
        return False
```
We want to skip fluentd tests for ubuntu-noble, regardless of arch.

File: internal/buildscripts/packaging/installer/install.sh [hunk 2]
```diff
@@ -692,6 +692,9 @@ install() {
       fi
       apt-get -y update
       install_apt_package "splunk-otel-collector" "$collector_version"
+      if [ "$distro_codename" = "noble" ]; then
+          $td_agent_version = ""
+      fi
```
Comment (jeffreyc-splunk) [inline internal/buildscripts/packaging/installer/install.sh:697]:
```suggestion
```
You can update the `fluentd_supported` function for `ubuntu` in https://github.com/signalfx/splunk-otel-collector/blob/main/internal/buildscripts/packaging/installer/install.sh#L1036 with something like:
```
if [ "$distro_version" = "24.04" ]; then
  return 1
fi
```
Everything else will then be automatically handled for you.

File: internal/buildscripts/packaging/tests/instrumentation/images/deb/Dockerfile.ubuntu-noble [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+RUN bash /opt/setup-tomcat.sh
+
+ARG NODE_VERSION=v16
+COPY instrumentation/setup-express.sh /opt
+RUN bash /opt/setup-express.sh
+
+COPY instrumentation/setup-dotnet.sh /opt
+RUN if [ "$TARGETARCH" = "amd64" ]; then \
+    apt-get update && \
+    apt-get install -yq libicu66 && \
```
Comment (jeffreyc-splunk) [inline internal/buildscripts/packaging/tests/instrumentation/images/deb/Dockerfile.ubuntu-noble:35]:
```suggestion
    apt-get install -yq libicu74 && \
```

**Review by jeffreyc-splunk — APPROVED** [review]
Thanks @samiura!  Please work with the docs team to include Ubuntu 24.04 to the list of supported distros for the installer script, and note that fluentd is not supported.

**Pull request merged**